### ISSUE TITLE
Upgrade from 4.4 to 4.5 or 4.6 breaks functioning Windows 10 VMs

### ISSUE THREAD

**Issue by stragu** [quickemu-project/quickemu#654]
#### Expected behaviour

Windows 10 VM continues to boot after upgade from 4.4 to 4.5 or 4.6

#### Actual behaviour

Windows VM alternates between repair mode and the "INACCESSIBLE BOOT DEVICE" blue error screen when booted.

Reported by two other users too:
@matthewadie in https://github.com/quickemu-project/quickemu/issues/572#issuecomment-1346934813
@antonc42 in https://github.com/quickemu-project/quickemu/issues/572#issuecomment-1348787171

#### Steps to reproduce the behaviour

Have a previously functioning VM, upgrade from 4.4 to 4.5 or 4.6

Config:
```
#!/usr/bin/quickemu --vm
guest_os="windows"
disk_img="windows-10/disk.qcow2"
iso="windows-10/Win10_22H2_EnglishInternational_x64.iso"
fixed_iso="windows-10/virtio-win.iso"
```

Launched with: `quickemu --vm ~/windows-10.conf --display spice`

Tried:
- with or without ignoring unhandled MSRs
- secure boot on with `secureboot="on"`: black screen
- using TPM with `tpm=on`: same result

Workaround was to use an older quickemu deb to rollback to 4.4 but that isn't available on the PPA anymore.

#### Quickemu output

```
Quickemu 4.6 using /usr/bin/qemu-system-x86_64 v6.0.0
 - Host:     Ubuntu 20.04.5 LTS running Linux 5.15 (xxx)
 - CPU:      Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
 - CPU VM:   1 Socket(s), 2 Core(s), 2 Thread(s), 4G RAM
 - BOOT:     EFI (Windows), OVMF (/usr/share/OVMF/OVMF_CODE.fd), SecureBoot (off).
 - Disk:     windows-10/disk.qcow2 (64G)
 - CD-ROM:   windows-10/virtio-win.iso
 - Display:  SPICE, qxl-vga, GL (on), VirGL (off)
 - ssh:      On host:  ssh user@localhost -p 22220
 - SPICE:    On host:  spicy --title "windows-10" --port 5930 --spice-shared-dir /home/stragu/Public
 - WebDAV:   On guest: dav://localhost:9843/
 - smbd:     On guest: smb://10.0.2.4/qemu
 - Network:  User (virtio-net)
 - Monitor:  On host:  nc -U "windows-10/windows-10-monitor.socket"
             or     :  socat -,echo=0,icanon=0 unix-connect:windows-10/windows-10-monitor.socket
 - Serial:   On host:  nc -U "windows-10/windows-10-serial.socket"
             or     :  socat -,echo=0,icanon=0 unix-connect:windows-10/windows-10-serial.socket
 - Process:  Starting /home/stragu/windows-10.conf as windows-10 (73022)
 - Viewer:   spicy --title "windows-10" --port "5930" --spice-shared-dir "/home/stragu/Public" "" >/dev/null 2>&1 &
```

#### Linux Distribution & Kernel

```bash
LSB Version:	core-11.1.0ubuntu2-noarch:printing-11.1.0ubuntu2-noarch:security-11.1.0ubuntu2-noarch
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.5 LTS
Release:	20.04
Codename:	focal

Linux xxx 5.15.0-60-generic #66~20.04.1-Ubuntu SMP Wed Jan 25 09:41:30 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by TuxVinyards** [comment]
"Workaround was to use an older quickemu deb to rollback to 4.4 but that isn't available on the PPA anymore."

https://github.com/quickemu-project/quickemu/releases/tag/4.4

Paste the quickemu file from this zip into /usr/bin

Or:

![win-rc-2](https://user-images.githubusercontent.com/3956806/220627745-61d94ebc-fe09-4d1c-850f-ae8800da6512.jpg)

https://github.com/TuxVinyards/quickemu-mod

**Comment by stragu** [comment]
Thank you for the workaround, @TuxVinyards ! Pasting the 4.4 `quickemu` file into `/usr/bin` made the VM usable again. An update was automatically unapplied and then it booted normally.

**Comment by stueken** [comment]
Same problem here. Updating from 4.4 to 4.6 destroyed my Windows VM.
Just moving back to 4.4 with `git reset --hard 4.4` solved the problem for me.
Hope this problem gets solved with the next version.

**Comment by AkechiShiro** [comment]
There should be a solution that should be worked on, in order to achieve safe updates and not break on users, I'm totally unfamiliar with the code base but I believe something like this should be considered by maintainers.

**Comment by TuxVinyards** [comment]
@AkechiShiro if you use the quickemu-wrap script, you now have the possibility to record the virtual hardware profile. As we all know, Windows doesn't like it if you install lots of new hardware all at once. The script allows you add extra profiles as you go too. 

The other thing you can do with quickemu-wrap is to mix and match different quickemu versions with different versions of quickget.  This means you can use a version of quickemu that stays compatible with your Windows VM but have all the benefits of the latest quickget, the part of the codebase that has a lot more updates added to it.

I have nearly finished the dev work on qqX which will  replace quickemu-wrap. This has a lot more features, much more desktop integration and an update checker ....

**Comment by AkechiShiro** [comment]
Thanks @TuxVinyards could you link the repo of qqX ? Or it's a script embedded in quickemu ?

**Comment by TuxVinyards** [comment]
Check my profile ... 😉

**Comment by flexiondotorg** [comment]
Fixed via #836
